app/controllers/pages_controller.rb

```diff
@@ -1,6 +1,6 @@
 class PagesController < ApplicationController
   before_action :check_user_has_permission
-  skip_before_action :clear_draft_questions_data, except: %i[index move_page]
+  before_action :clear_draft_questions_data, only: %i[index move_page]
```

Comment:
I like this change, it is easier to understand for me. I'm pretty sure it means the same thing now you've moved the action method?

Reply:
I think its clearer to me, but there is a change in when the `clear_draft_questions_data` is invoked which I would welcome you're opinion on. 

When it was in the ApplicationController is was being invoked whenever a user visited almost any page (it was excluded in the heartbeat controller). This I believe was causing a problem mentioned above under "second problem". If a user was editing a question and happened to visit any other page in another tab it would clear the draft question they were working on which doesn't seem right and I think causes some of the [draft question exceptions](https://govuk-forms.sentry.io/issues/4200026265/?project=6576190). It makes sense to me that we only clear draft questions, when they save their question, visit the new method (or as it was also on `move_page`, not sure why but that was already there). It's also clearer to me to have this in the pages controller because that's its concern. It was confusing when I looked at it originally in the ApplicationController and wondering why we're clearing draft questions on very unrelated pages, like viewing the MOU for example.

To be honest, I would err towards that we explicitly call `clear_draft_questions_data` from the appropriate controller methods when there are few specific reasons to call it (unlike ensuring authentication which is wide spread and safer to `before_action` as the ApplicationController level). We currently have a mix because we call it explicitly in the update questions (https://github.com/alphagov/forms-admin/blob/main/app/controllers/pages/questions_controller.rb#L48). This to me is simpler to understand what that controller method does, but using the `before_action` with `only` seems the more Railsy way :)

Reply:
Sorry, yes, I meant it's the same within the PagesController class and subclasses. I agree there's not much need to clear the draft question in pages that aren't related to editing a particular form. We could end up with some storage leaking over time, but I don't think it would be that much of an issue (famous last words).

Personally I'm not too bothered either way about using `before_action` or not, but googling for 'ruby rails when not to use before_action' suggests that there a different schools of thought out there...

Reply:
yeah I'm happy to leave it with `before_action` if that's the norm. Ok to revolve this conversation?

Reply:
Resolved :D